File: ProjetoJava/src/associacao.java
import java.util.ArrayList;
import java.util.List;

class Professor {
    String nome;
    List<Escola> escolas = new ArrayList<>();

    public Professor(String nome) {
        this.nome = nome;
    }

    public void associarEscola(Escola escola) {
        escolas.add(escola);
    }
}

class Escola {
    String nome;
    List<Professor> professores = new ArrayList<>();

    public Escola(String nome) {
        this.nome = nome;
    }

    public void adicionarProfessor(Professor professor) {
        professores.add(professor);
    }

    public static void main(String[] args) {
        Escola escola = new Escola("Escola Primária");
        Professor professor = new Professor("Maria");
        escola.adicionarProfessor(professor);
        professor.associarEscola(escola);

        System.out.println(professor.nome + " leciona na " + escola.nome);
    }
}

File: GoLangProject/associacao.go
package main

import "fmt"

type Professor struct {
	Nome    string
	Escolas []*Escola
}

type Escola struct {
	Nome        string
	Professores []*Professor
}

func (p *Professor) AdicionarEscola(e *Escola) {
	p.Escolas = append(p.Escolas, e)
	e.Professores = append(e.Professores, p)
}

func main() {
	escola := Escola{Nome: "Escola Primária"}
	professor := Professor{Nome: "Maria"}
	professor.AdicionarEscola(&escola)

	fmt.Printf("%s leciona na %s\n", professor.Nome, escola.Nome)
}
